### PR TITLE
fix regions update method

### PR DESCRIPTION
**Pull request by entonbiba**
**to do:**
- fix cursor on on drag
- create method for 
```
if(!currentDrag || !currentResize){
    this.bindEvents(regionEl);
}
```

enables resize and drag after user has created regions with params set to false. #1062

```
  wavesurfer.addRegion({
    id: 1,
    start: 2, // time in seconds
    end: 5, // time in seconds
    color: "hsla(100, 100%, 30%, 0.1)",
    drag: false, <---
    resize: false <---
  });
```

before fix:
http://codepen.io/entonbiba/pen/ZKKadq
```
  wavesurfer.regions.list[1].update({ resize: "true", drag: "true" });
// won't work since .update doesn't do anything since already set to false in addRegion. 
```

after fix:
http://codepen.io/entonbiba/pen/zwZyOg
```
  wavesurfer.regions.list[1].update({ resize: "true", drag: "true" });
// works as expected
```

### CODE REVIEW

File: plugin/wavesurfer.regions.js [hunk 1]
```diff
@@ -182,6 +184,36 @@ WaveSurfer.Region = {
...
+            this.style(handleRight, css);
+            this.style(handleRight, {
+                left: '100%'
+            });
+            this.element = this.wrapper.appendChild(regionEl);
+        }
+
+        if(!currentDrag || !currentResize){
+            this.bindEvents(regionEl);
+        }
```
Comment (katspaugh) [inline plugin/wavesurfer.regions.js]: I think it's better to extract this part of the [`render` function](https://github.com/katspaugh/wavesurfer.js/blob/master/plugin/wavesurfer.regions.js#L213-L257) into a separate function and call it from both places passing either a freshly created region element or an existing element.

Comment (entonbiba) [inline plugin/wavesurfer.regions.js]: @katspaugh yes that sounds good, I'll do that

**Comment by agamemnus** [comment]
Fundamentally, this needed some more changes: https://github.com/katspaugh/wavesurfer.js/pull/1103.

**Comment by agamemnus** [comment]
There are really two ways to do this.

1) Adding/removing the elements themselves.
2) Updating whether they are displayed via the DOM.
3) Updating the logic to say whether mouse events on them should have any effect.

They all have pros and cons...

**Comment by thijstriemstra** [comment]
Please open a new pull request against the latest master branch, this one diverged too much.

**Pull request closed without merging**